SDKSampleApp/ViewModel/QRScannerViewModel.swift

```diff
@@ -0,0 +1,215 @@
+            let queue = DispatchQueue.global(qos: .background)
+            queue.async { [weak self] in
+                self?.session.startRunning()
+            }
+            isScanning = true
+        }
+    }
+    
+    func tryToStopSession() {
+        if session.isRunning == true {
+            session.stopRunning()
+        }
+    }
+    
+    func createSession(
+        rectOfInterest: CGRect
+    ) -> AVCaptureSession? {
+        guard let captureDevice = AVCaptureDevice.default(for: .video) else { return nil }
+        
+        do {
+            let input = try AVCaptureDeviceInput(device: captureDevice)
+            session.addInput(input)
+        } catch {
+            print(error.localizedDescription)
+            return nil
+        }
+        
+        let output = createAVCaptureMetadataOutput(rectOfInterest: rectOfInterest)
+        session.addOutput(output)
+        output.metadataObjectTypes = [.qr]
+        
+        return session
+    }
+    
+    func createAVCaptureMetadataOutput(
+        rectOfInterest: CGRect
+    ) -> AVCaptureMetadataOutput {
+        let screenWidth = UIScreen.main.bounds.width
+        let screenHeight = UIScreen.main.bounds.height
+        
+        let output = AVCaptureMetadataOutput()
+        
+        output.setMetadataObjectsDelegate(self, queue: DispatchQueue.main)
+        
+        output.rectOfInterest = CGRect(
+            x: rectOfInterest.origin.y / screenHeight,
+            y: rectOfInterest.origin.x / screenWidth,
+            width: rectOfInterest.size.height / screenHeight,
+            height: rectOfInterest.size.width / screenWidth
+        )
+        return output
+    }
+    
+}
+
+// MARK: - AVFoundation QR
+extension QRScannerViewModel: AVCaptureMetadataOutputObjectsDelegate {
+    internal func metadataOutput(
+        _ output: AVCaptureMetadataOutput,
+        didOutput metadataObjects: [AVMetadataObject],
+        from connection: AVCaptureConnection
+    ) {
+        if isScanning {
+            isScanning = false
+            if let metadataObject = metadataObjects.first as? AVMetadataMachineReadableCodeObject {
+                if metadataObject.type == .qr {
+                    if let qrCodeData = metadataObject.stringValue {
+                        let parameters = extractURLParameters(from: qrCodeData)
+                        navigateWithQRParams(qrParams: parameters)
+                        session.stopRunning()
+                        AudioServicesPlaySystemSound(SystemSoundID(kSystemSoundID_Vibrate))
+                    }
+                }
+            }
+        }
+    }
+}
+
+// MARK: - private functions
+extension QRScannerViewModel {
+    
+    func extractURLParameters(from qrCodeData: String) -> QRCodeURLParameters {
+        var parameters = QRCodeURLParameters()
+
+        let cleanedData = qrCodeData.replacingOccurrences(of: jsPlayerURL, with: "")
+        
+        let parameterPairs = cleanedData.components(separatedBy: "&")
```

Comment:
Wouldn't it be better to use `URLComponents` `queryItems` and `URL` instead of working with `String`?